diff --git a/Services/UI/ImageCacheService.qml b/Services/UI/ImageCacheService.qml
index 9da04cda..6c57fc4f 100644
--- a/Services/UI/ImageCacheService.qml
+++ b/Services/UI/ImageCacheService.qml
@@ -274,7 +274,7 @@ Singleton {
     const doubleWidth = width * 2;
     const doubleHeight = height * 2;
 
-    const command = `magick -define jpeg:size=${doubleWidth}x${doubleHeight} '${srcEsc}' -auto-orient -thumbnail '${width}x${height}^' -quality 95 '${dstEsc}'`;
+    const command = `wallpaper resize --size ${width}x${height} ${srcEsc} ${dstEsc}`
 
     runProcess(command, cacheKey, outputPath, sourcePath);
   }
