From 90c86c2078cdd2859474e48b647ddc3740dc0533 Mon Sep 17 00:00:00 2001
From: Lin Xianyi <iynaix@gmail.com>
Date: Fri, 14 Jul 2023 21:02:51 +0800
Subject: [PATCH] Disable key repeat timer

---
 src/wl_window.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/wl_window.c b/src/wl_window.c
index 0c2221a..43d558c 100644
--- a/src/wl_window.c
+++ b/src/wl_window.c
@@ -505,7 +505,7 @@ static void on_global(void *data, struct wl_registry *registry, uint32_t id,
 
   if (!strcmp(interface, "wl_compositor")) {
     version = imv_min(version, 4);
-    window->wl_compositor = 
+    window->wl_compositor =
       wl_registry_bind(registry, id, &wl_compositor_interface, version);
   } else if (!strcmp(interface, "xdg_wm_base")) {
     version = imv_min(version, 2);
@@ -793,7 +793,7 @@ struct imv_window *imv_window_create(int width, int height, const char *title)
   create_window(window, width, height, title);
 
   struct sigevent timer_handler = {
-    .sigev_notify = SIGEV_THREAD,
+    .sigev_notify = SIGEV_NONE,
     .sigev_value.sival_ptr = window,
     .sigev_notify_function = on_timer,
   };
-- 
2.41.0

