diff --git a/niri-config/src/appearance.rs b/niri-config/src/appearance.rs
index eba265b0..274bdc01 100644
--- a/niri-config/src/appearance.rs
+++ b/niri-config/src/appearance.rs
@@ -432,7 +432,7 @@ pub struct WorkspaceShadowPart {
     #[knuffel(child, unwrap(argument))]
     pub softness: Option<FloatOrInt<0, 1024>>,
     #[knuffel(child, unwrap(argument))]
-    pub spread: Option<FloatOrInt<-1024, 1024>>,
+    pub spread: Option<FloatOrInt<-4096, 4096>>,
     #[knuffel(child)]
     pub color: Option<Color>,
 }
@@ -649,7 +649,7 @@ pub struct ShadowRule {
     #[knuffel(child, unwrap(argument))]
     pub softness: Option<FloatOrInt<0, 1024>>,
     #[knuffel(child, unwrap(argument))]
-    pub spread: Option<FloatOrInt<-1024, 1024>>,
+    pub spread: Option<FloatOrInt<-4096, 4096>>,
     #[knuffel(child, unwrap(argument))]
     pub draw_behind_window: Option<bool>,
     #[knuffel(child)]
