diff --git a/src/layout/tile.rs b/src/layout/tile.rs
index bc39db6d..b1ae438b 100644
--- a/src/layout/tile.rs
+++ b/src/layout/tile.rs
@@ -1216,6 +1216,7 @@ impl<W: LayoutElement> Tile<W> {
             .chain(rounded_corner_damage)
             .chain(window_surface.into_iter().flatten());
 
+        /*
         let elem = (fullscreen_progress > 0.).then(|| {
             let alpha = fullscreen_progress as f32;
 
@@ -1258,6 +1259,7 @@ impl<W: LayoutElement> Tile<W> {
             }
         });
         let rv = rv.chain(elem);
+        */
 
         let elem = self.visual_border_width().map(|width| {
             self.border
