diff --git a/pqiv.c b/pqiv.c
index 4ed495a..2403f60 100644
--- a/pqiv.c
+++ b/pqiv.c
@@ -8215,12 +8215,6 @@ int main(int argc, char *argv[]) {
 		#endif
 	#endif
 
-	#if GTK_CHECK_VERSION(3, 10, 0)
-		// Always prefer x11 over Wayland. Wayland rendering has several issues, in particular
-		// windows have an undesired black margin around them.
-		gdk_set_allowed_backends("x11,*");
-	#endif
-
 	#if defined(GDK_WINDOWING_X11)
 		XInitThreads();
 	#endif
