diff --git a/pkgs/by-name/aw/awakened-poe-trade/package.nix b/pkgs/by-name/aw/awakened-poe-trade/package.nix
index fcc8d734b5f37..86009f10cc628 100644
--- a/pkgs/by-name/aw/awakened-poe-trade/package.nix
+++ b/pkgs/by-name/aw/awakened-poe-trade/package.nix
@@ -9,6 +9,7 @@
   libxt,
 
   nix-update-script,
+  commandLineArgs ? [ ],
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -43,7 +44,7 @@ stdenv.mkDerivation (finalAttrs: {
     cp -a ${finalAttrs.passthru.appImageContents}/usr/share/icons $out/share
 
     substituteInPlace $out/share/applications/awakened-poe-trade.desktop \
-      --replace-fail 'Exec=AppRun' 'Exec=awakened-poe-trade'
+      --replace-fail 'Exec=AppRun' 'Exec=awakened-poe-trade ${lib.escapeShellArgs commandLineArgs}'
 
     runHook postInstall
   '';
@@ -51,6 +52,9 @@ stdenv.mkDerivation (finalAttrs: {
   postFixup = ''
     makeWrapper ${lib.getExe electron} $out/bin/awakened-poe-trade \
       --add-flags $out/share/awakened-poe-trade/resources/app.asar \
+      ${
+        lib.optionalString (commandLineArgs != [ ]) "--add-flags ${lib.escapeShellArgs commandLineArgs}"
+      } \
       --prefix LD_LIBRARY_PATH : "${
         lib.makeLibraryPath [
           libxtst
